geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -143,12 +145,50 @@ protected class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends Duplicati
     }
   }
 
+  private def handleInArrayEquals(filter: PropertyIsEqualTo, extraData: AnyRef): Option[Filter] = {
+    var inArrayFunction: InArrayFunction = null
+    var hasTrue: Boolean = false
+
+    def inspectExpression(expression: Expression) = {
+      expression match {
+        case function: InArrayFunction =>
+          inArrayFunction = function
+        case literal: Literal if literal.getValue.equals(java.lang.Boolean.TRUE) =>
```

Comment:
is it possible that literal.getValue might be null? i'm not sure... it would be safer to flip the `.equals` to `java.lang.Boolean.TRUE.equals(...)` or just use scala `==`, which both should just return false for a null but not throw an NPE

Reply:
+1.